Example manifests for VMWare

bvankampen opened the pull request:
kind/examples

**What this PR does / why we need it**:

Add example manifiest for provisioning RKE2 clusters on VMware.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Missing examples for VMware

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits into logical changes
- [X] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

alexander-demicev reviewed (approved):
Looks good to me, we will have to add instructions on uploading machine image later when we have some, similar to this doc https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/getting_started.md#uploading-the-machine-images

bvankampen commented:
@alexander-demicev Yes no problem, they will be mostly the same as the VMware instructions. I will add them later. Also I think I removed your approval when I added my signoff.

belgaied2 reviewed (approved):
/LGTM
